src/roller.js

```diff
@@ -93,28 +94,31 @@ class LMRTFYRoller extends Application {
         this.element.find(".lmrtfy-ability-save").click(this._onAbilitySave.bind(this))
         this.element.find(".lmrtfy-skill-check").click(this._onSkillCheck.bind(this))
         this.element.find(".lmrtfy-custom-formula").click(this._onCustomFormula.bind(this))
-        this.element.find(".lmrtfy-initiative").click(this._onInitiative.bind(this))
-        this.element.find(".lmrtfy-death-save").click(this._onDeathSave.bind(this))
+        if(LMRTFY.specialRolls.includes('initiative')) {
+            this.element.find(".lmrtfy-initiative").click(this._onInitiative.bind(this))
+        }
+        if(LMRTFY.specialRolls.includes('deathsave')) {
+            this.element.find(".lmrtfy-death-save").click(this._onDeathSave.bind(this))
+        }
+        if(LMRTFY.specialRolls.includes('perception')) {
+            this.element.find(".lmrtfy-perception").click(this._onPerception.bind(this))
+        }
     }
 
     _makeRoll(event, rollMethod, ...args) {
         let fakeEvent = {}
-        if (this.advantage === 0) {
-            fakeEvent.shiftKey = !LMRTFY.defaultQuery;
-            fakeEvent.altKey = false;
-            fakeEvent.ctrlKey = false;
-        } else if (this.advantage === 1) {
-            fakeEvent.shiftKey = false;
-            fakeEvent.altKey = true;
-            fakeEvent.ctrlKey = false;
-        } else if (this.advantage === -1) {
-            fakeEvent.shiftKey = false;
-            fakeEvent.altKey = false;
-            fakeEvent.ctrlKey = true;
-        } else if (this.advantage === 2) {
-            fakeEvent.shiftKey = LMRTFY.defaultQuery;
-            fakeEvent.altKey = false;
-            fakeEvent.ctrlKey = false;
+        switch(this.advantage) {
+            case -1: 
+                fakeEvent = LMRTFY.disadvantageRollEvent;
+                break;
+            case 1:
+                fakeEvent = LMRTFY.advantageRollEvent;
+                break;
+            case 2: 
+                fakeEvent = LMRTFY.modifiedRollEvent;
+                break;
+            default:
+                fakeEvent = LMRTFY.defaultRollEvent;
```

Comment:
I don't get the `modifiedRoll`, should just be named `queryRoll` which makes it more understandable. Also, `defaultRoll` should really be `normalRoll` since the default here is tied to the "Normal Roll" modifier.
I also personally like to list all cases, so I would do :
```js
case 0:
default:
```
Just to make it obvious that case 0 (normal roll) is taken into account. Usually `default` is left as "fallback in case everything else fails", so reading this in a few weeks, it would feel to me like default would basically never be called unless there's some unexpected error and I wouldn't realize that "0" is the actual default and is a valid value.

Reply:
Hahaha... I had queryRoll before, I avoided it this time ;-)

Resolved and agreed on case statement for readability. Done.